File: wallet-kit/src/main/java/bitcoin/walllet/kit/network/message/VerAckMessage.java
package bitcoin.walllet.kit.network.message;

/**
 * ACK for Version Message.
 *
 * @author liaoxuefeng
 */
public class VerAckMessage extends Message {

    public VerAckMessage() {
        super("verack");
    }

    public VerAckMessage(byte[] payload) {
        super("verack");
    }

    @Override
    protected byte[] getPayload() {
        return new byte[0];
    }

    @Override
    public String toString() {
        return "VerAckMessage()";
    }

}

File: wallet-kit/src/main/kotlin/bitcoin/wallet/kit/messages/VerAckMessage.kt
package bitcoin.wallet.kit.messages

/**
 * The 'verack' message is sent in response to the 'version' message.
 * It consists of just the message header.
 */
class VerAckMessage() : Message("verack") {

    constructor(payload: ByteArray) : this()

    override fun getPayload(): ByteArray {
        return ByteArray(0)
    }

    override fun toString(): String {
        return "VerAckMessage()"
    }
}
